### PR TITLE
[1860/2] Update stock movement to reflect new zigzag market behavior

### LINKED ISSUE #8972: [1862] Game that was previously working is now failing to parse previous actions.
We're sorry this game cannot be continued due to GameError: Blocking step Sell then Buy Shares cannot process action
Broken action: { "type": "pass", "entity": "I&B", "entity_type": "corporation", "created_at": [1674441517](tel:1674441517) }
Please [raise a bug report] and include this link (https://18xx.games/game/109255?action=204)

### LINKED ISSUE #8974: [1862] Error occurred in Merge.
https://18xx.games/game/113619

We're sorry this game cannot be continued due to GameError: Blocking step Merge cannot process action 594
Broken action: { "type": "pass", "entity": "L&H", "entity_type": "corporation", "id": 594, "created_at": 1678040668 }
Please raise a bug report and include this link ([https://18xx.games/game/113619?action=593](https://18xx.games/game/113619?action=593))

### PR DESCRIPTION
**Pull request by crericha**
Fixes #8974 
Fixes #8972

### CODE REVIEW

File: lib/engine/game/g_1860/game.rb [hunk 1]
```diff
@@ -743,8 +743,8 @@ def current_layer
         end
 
         def move_marker_to_bottom!(corporation)
-          @stock_market.move_left(corporation)
-          @stock_market.move_right(corporation)
+          @stock_market.move_down(corporation)
+          @stock_market.move_up(corporation)
```
Comment (roseundy) [inline lib/engine/game/g_1860/game.rb:747]: While I don't disagree with this change, it shouldn't have caused a failure (assume a move_right still reverses a move_left).

**Comment by roseundy** [comment]
Looks like all game-specific 'move_left' and 'move_right' calls have been fixed.

LGTM.

**Review by roseundy — APPROVED** [review]

File: lib/engine/game/g_1860/game.rb (hunk 1 above)
Comment (crericha) [reply lib/engine/game/g_1860/game.rb:747]: I didn't know what the behavior was doing, so I changed all of them. Now that you explained it, I agree it works either way.

**Comment by physic** [comment]
How can I tell when this is deployed? #8972 is still failing on the site.

**Comment by tobymao** [comment]
go to the about page, it will tell you the git hash

**Pull request merged**